types/user.ts

```diff
@@ -76,6 +77,33 @@ export class User extends LazyObject {
     });
   }
 
+  public async unfollowUser(other: User) {
+    // If we already have data, might as well run the free short-circuit check.
+    // We're going to run it anyways during the transaction, but if we can avoid it,
+    // might as well do it now.
+    if (this.hasData && !containsRef(this.following!, other)) return;
+
+    return runTransaction(db, async (transaction: Transaction) => {
+      const thisSnap = await transaction.get(this.docRef!);
+      const otherSnap = await transaction.get(other.docRef!);
+
+      this.initWithDocumentData(thisSnap.data()!);
+      other.initWithDocumentData(otherSnap.data()!);
+
+      if (!containsRef(this.following!, other)) return;
+
+      this.following = this.following!.filter((user) => user !== other);
```

Comment:
Not quite; it's more like
`user.docRef && refEqual(user.docRef, other.docRef!)`

Reply:
@JakeSteinebronn can you please explain what you mean again?

Reply:
So it's like we could have two User objects which refer to the same document. In this case, the === would say they're not equal, but we really should be deleting it. And in fact, it's overwhelmingly likely for this to be the case, where you got the "other" user object from somewhere outside of your own follow list. Furthermore, it is almost impossible for you to get your own user object from the other's list of followers. However, we should remove the *corresponding* user object from the respective arrays which refer to the same document.

Reply:
Also, I already wrote a function for this called removeRef, it's in common